### PR TITLE
chore(deps): Add 3rd party license file and CI checks

### PR DESCRIPTION
**Pull request by bruceg**
This adds a generated `LICENSE-3rdparty.csv` file mandated by DataDog's open source policy, as well as a CI step to check that the file is up to date.

### CODE REVIEW

**Comment by bruceg** [comment]
I set up this check in the line with all the other "Checks", but I could see it making sense to break out this and the other checks that are dependent on deps changing going in their own run.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -346,6 +346,12 @@ jobs:
       - name: Check cargo deny advisories/licenses
         if: needs.changes.outputs.dependencies == 'true' || needs.changes.outputs.deny == 'true'
         run: make check-deny
+      - name: Install the 3rd-party license tool
+        if: needs.changes.outputs.dependencies == 'true' || needs.changes.outputs.deny == 'true'
```
Comment (jszwedko) [inline .github/workflows/test.yml:350]:
```suggestion
        if: needs.changes.outputs.dependencies == 'true'
```

Doesn't seem like you'd need to run it on deny changes.

File: license-tool.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+"vrl-diagnostic" = { license = "MPL-2.0" }
+"vrl-parser" = { license = "MPL-2.0" }
+"vrl-tests" = { license = "MPL-2.0" }
+
+"ring-0.16.20" = { license = "Custom" }
+"rustls-webpki-0.100.1" = { license = "ISC-style" }
+"webpki-0.21.4" = { license = "ISC-style" }
+"webpki-0.22.0" = { license = "ISC-style" }
+"zerocopy-0.6.1" = { license = "BSD-2-Clause" }
+"zerocopy-derive-0.3.2" = { license = "BSD-2-Clause" }
```
Comment (jszwedko) [inline license-tool.toml:21]: Should each of these have a comment explaining?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -346,6 +346,12 @@ jobs:
       - name: Check cargo deny advisories/licenses
         if: needs.changes.outputs.dependencies == 'true' || needs.changes.outputs.deny == 'true'
         run: make check-deny
+      - name: Install the 3rd-party license tool
```
Comment (jszwedko) [inline .github/workflows/test.yml:349]: Usually we have this installation in the bootstrap file. Would it make sense to put this there?

File: license-tool.toml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[overrides]
+"backon" = { origin = "https://github.com/Xuanwo/backon" }
+"bollard-stubs" = { origin = "https://github.com/fussybeaver/bollard" }
+"openssl-macros" = { origin = "https://github.com/sfackler/rust-openssl" }
+"serde_nanos" = { origin = "https://github.com/caspervonb/serde_nanos" }
```
Comment (jszwedko) [inline license-tool.toml:5]: Why are these needed?

**Review by jszwedko — COMMENTED** [review]
Awesome! Just a few comments.

File: license-tool.toml (hunk 4 above)
Comment (bruceg) [reply license-tool.toml:5]: The sources don't actually include either a `repository` or `homepage` line in their respective metadata, so we can't automatically determine where they are actually hosted.

File: .github/workflows/test.yml (hunk 3 above)
Comment (bruceg) [reply .github/workflows/test.yml:349]: I could go either way. It's _only_ needed for this one check, so 🤷🏻

File: .github/workflows/test.yml (hunk 1 above)
Comment (bruceg) [reply .github/workflows/test.yml]: Ah, no, I just copied from the previous line, fixed in 841659fb8.

File: license-tool.toml (hunk 2 above)
Comment (bruceg) [reply license-tool.toml]: Oh, yeah, probably. Fixed in 4ed90410c

File: license-tool.toml (hunk 4 above)
Comment (spencergilbert) [reply license-tool.toml:5]: Worth PRing these few upstream to reduce the manual work here?

**Review by spencergilbert — COMMENTED** [review]
Currently failing the CI check, and one thought on the `license-tool.toml`.

Which as I say it, could be nice to move it into a subdirectory - we have a lot of files in the root.

File: .github/workflows/test.yml (hunk 3 above)
Comment (jszwedko) [reply .github/workflows/test.yml:349]: True, though the bootstrap scripts are also run as part of `make environment` which installs needed development tools.

File: license-tool.toml (hunk 4 above)
Comment (bruceg) [reply license-tool.toml:5]: Yeah, probably.

**Comment by bruceg** [comment]
> Which as I say it, could be nice to move it into a subdirectory - we have a lot of files in the root.

Could do this, but it'll need an option to the license tool. The filename is hard-coded. Which, of course, is not the greatest situation, particularly since it is undocumented…

**Review by jszwedko — APPROVED** [review]

**Comment by jszwedko** [comment]
Thanks for the updates!
